sky/shell/ios/sky_surface.mm

```diff
@@ -146,6 +167,12 @@ - (void)connectToEngineAndLoad {
 
   NSString *endpoint = self.skyInitialBundleURL;
   if (endpoint.length > 0) {
+#ifndef NDEBUG
```

Comment:
This leaks the watcher. You could cancel the watcher and release its reference in the `didMoveToWindow` override where `window == nil`